src/ui/src/builder/settings/BuilderSettingsArtifactAPITriggerDetails.vue

```diff
@@ -0,0 +1,171 @@
+                        Copy cURL command
+                    </WdsButton>
+                </li>
+                <li class="BuilderSettingsArtifactAPITriggerDetails__step" id="stepPaste">
+                    <p class="BuilderSettingsArtifactAPITriggerDetails__step__hint">
+                        Once the job has run, paste the result here:
+                    </p>
+                    <WdsTextareaInput
+                        v-model="apiOutput"
+                        placeholder="Paste cURL output here"
+
+                    />
+                </li>
+                <li class="BuilderSettingsArtifactAPITriggerDetails__step" id="stepPoll">
+                    <p class="BuilderSettingsArtifactAPITriggerDetails__step__hint">
+                        Lastly, paste the poll async into your terminal to retrieve your artifact
+                    </p>
+                    <WdsButton variant="tertiary" size="small" @click="copyPoll" :disabled="isPollDisabled">
+                        <i class="material-symbols-outlined">{{ isPollCopied ? 'check' : 'content_copy' }}</i>
+                        Copy poll async
+                    </WdsButton>
+                </li>
+            </ol>
+        </div>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { computed, inject, ref } from 'vue';
+import injectionKeys from '@/injectionKeys';
+import WdsButton from '@/wds/WdsButton.vue';
+import WdsTabs, { WdsTabOptions } from '@/wds/WdsTabs.vue';
+import WdsTextareaInput from '@/wds/WdsTextareaInput.vue';
+import WdsFieldWrapper from '@/wds/WdsFieldWrapper.vue';
+import BuilderFieldsCode from './BuilderFieldsCode.vue';
+import { useButtonClipboard } from '../useButtonClipboard';
+import WdsTitle2 from '@/wds/WdsTitle2.vue';
+import { FieldType } from '@/writerTypes';
+
+const wf = inject(injectionKeys.core);
+const ssbm = inject(injectionKeys.builderManager);
+
+const component = computed(() => wf.getComponentById(ssbm.firstSelectedId.value));
+const blueprintId = computed(() => component.value?.parentId ?? '');
+
+const componentDefinition = computed(() => {
+    if (!component.value) return null;
+    return wf.getComponentDefinition(component.value.type);
+});
+
+const artifactFields = computed(() => {
+    const allFields = componentDefinition.value?.fields ?? {};
+    return Object.entries(allFields).filter(([, f]) => f.isArtifactField);
+});
+
+const baseURL =
+    import.meta.env.VITE_WRITER_BASE_URL ??
+    window.location.origin + window.location.pathname;
+const agentId = import.meta.env.VITE_WRITER_APP_ID ?? '';
+const apiKey = import.meta.env.VITE_WRITER_API_KEY ?? '';
+
+const curlCreate = computed(() =>
+    `curl --location --request POST '${baseURL}/v1/agents/${agentId}/blueprints/${blueprintId.value}/jobs' \\` +
+    `\n--header 'Authorization: Bearer ${apiKey}' \\` +
+    `\n--header 'Content-Type: application/json' --data '{"inputs": []}'`
+);
```

Comment:
You can do this. The `VITE_*` env are data passed at build time (not at runtime).


```suggestion
const apiKey = import.meta.env.VITE_WRITER_API_KEY ?? '';

const curlCreate = computed(() =>
    `curl --location --request POST '${baseURL}/v1/agents/${wf.writerAppId}/blueprints/${blueprintId.value}/jobs' \\` +
    `\n--header 'Authorization: Bearer ${apiKey}' \\` +
    `\n--header 'Content-Type: application/json' --data '{"inputs": []}'`
);
```